### DISCUSSION TITLE
resource_account upgrade failed

### DISCUSSION THREAD

**Discussion by ctb0k33** [aptos-labs/aptos-developer-discussions#191] [Questions] [answered]
### Discord user ID

ctb0k33

### Describe your question in detail.

I try to upgrade a module that already deploy in a resource account but it's failed.
resource-account address: 0xa69b6d7d0651aa73562d629d204e90bb3f51c6d49ec4038b07832fc888ece0d1
module: hello
network: devnet
Command: aptos move run --json-file upgrade.json 

Here is my function_id in upgrade.json file:
"function_id": "0xa69b6d7d0651aa73562d629d204e90bb3f51c6d49ec4038b07832fc888ece0d1::hello::upgrade",

### What error, if any, are you getting?

"Simulation failed with status: FUNCTION_RESOLUTION_FAILURE\nExecution failed with message: Cannot find ModuleId { address: a69b6d7d0651aa73562d629d204e90bb3f51c6d49ec4038b07832fc888ece0d1, name: Identifier(\"hello\") }::IdentStr(\"upgrade\") in cache"

### What have you tried or looked at?  Or how can we reproduce the error?

I currently follow this tutorial: https://aptos.dev/tutorials/programmatic-upgradeable-module/#how-to-publish-modules-to-a-resource-account

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

N/A

**Comment by 0x-j** [comment, marked as answer]
Hi we no longer recommend developers to deploy module under resource acocunt, instead we suggest deploying under object as it's easier. Checkout this example on deploying and upgrading module in object.

https://github.com/aptos-labs/move-by-examples/tree/main/simple-todo-list/aptos/move

**Reply by ctb0k33** [reply]
Thanks for answer @0xaptosj . However, i have a question. Can i transfer the permission of update contract for another address with object deploy method.

**Reply by 0x-j** [reply]
Yes! You simply transfer the object to another address. I wrote this move script that transfer the object contains contract from testnet-profile-1 to testnet-profile-2, then i run upgrade.sh with testnet-profile-2 and it works.

https://github.com/0xaptosj/playground/blob/2ff3791b01331e4d0f1e2f36009dee3fc935b933/move/scripts/transfer_object.move

**Reply by ctb0k33** [reply]
that's work for me, thanks for your support !
